### PR TITLE
ffmpeg 4.2.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by lembacon** [comment]
```
Error: 1 problem in 1 formula detected
ffmpeg:
  * Non-executables were installed to "/usr/local/opt/ffmpeg/bin"
    The offending files are:
      /usr/local/opt/ffmpeg/bin/python
```

**Comment by robertleeblairjr** [comment]
I believe the failure is the python binding to mpv when it's built/compiled. You could submit a bug report to mpv on GH.
[https://github.com/mpv-player/mpv/issues](url)

**Comment by retokromer** [comment]
According to @lachs0r (mvp):

>Not our fault.

**Comment by robertleeblairjr** [comment]
> According to @lachs0r (mpv):
> 
> > Not our fault.

According to mpv compile page, they don't support linking against any version of FFmpeg other than the one they're currently using in the project, which is 4.1.x. So, this won't be resolved until mpv updates the project with using FFmpeg 4.2. However, I believe there is going to be a problem with having the commit because FFmpeg 4.2 won't be in the repo but the new version of mpv which requires it will want to build against it.

Copy/paste from mpv project page.

>FFmpeg ABI compatibility
mpv does not support linking against FFmpeg versions it was not built with, even if the linked version is supposedly ABI-compatible with the version it was compiled against. Expect malfunctions, crashes, and security issues if you do it anyway.
The reason for not supporting this is because it creates far too much complexity with little to no benefit, coupled with absurd and unusable FFmpeg API artifacts.
Newer mpv versions will refuse to start if runtime and compile time FFmpeg library versions mismatch.

**Comment by robertleeblairjr** [comment]
The homebrew formula for 'mpv' is pulling an archive of source that's not been updated since October 2018. If homebrew is going to build 'mpv' formula, then it would be better to do a build against mpv project's master branch which has updates to 'waf' whereby it has patches from the version included in the *.tar.gz source file this formula is downloading. I believe the problem would be resolved if this homebrew formula pulled the branch, which may not be possible, instead of an outdated archive on the mpv GitHub page.

**Comment by robertleeblairjr** [comment]
If homebrew is going to use outdated project compiled source files, then they need to have something in place which will not keep requirement/dependency formula from not being updated by those formula which depend on it. In this situation, mpv project is claiming not their fault because master branch builds just fine with FFmpeg 4.2, along with python and other build requirements. The project doesn't provide source archived releases very often and there have been close to 400 commits/fixes since the 0.29.1 package on master. I would submit a PR for updating the mpv formula to git pull master for the mpv project but this goes against their Acceptable Formula stipulation. Although, they've made an exception for certain formula such as 'pip'. I believe they could make an exception for mpv so that FFmpeg, which should take precedence, can not conflict with it.
@SMillerDev  Can you take a look at if mpv formula can be an exception for pulling master branch since the mpv project rarely updates their releases for source? It's conflicting with FFmpeg formula being updated. Thanks.

**Comment by SMillerDev** [comment]
I would be very hesitant to make exceptions here. Can you at least request that they tag a release?

**Comment by robertleeblairjr** [comment]
> I would be very hesitant to make exceptions here. Can you at least request that they tag a release?

Went to mpv project to file an issue to see if they would update. @retokromer was on it. Thanks for the reply.

**Comment by SMillerDev** [comment]
Got a link?

**Comment by robertleeblairjr** [comment]
@SMillerDev Can you look at making an exception or just having mpv removed as formula and available only as a cask? I believe the mpv project, at the moment, is not going to update their snapshot releases which this formula relies on. So, who knows if another minor revision will be released, not likely, or when the major version will be released. A maintainer at the project stated 0.30 release is unknown. From the conversation, it's most likely not any time soon. I would think that either mpv would need to be removed or something changed so as not to conflict with FFmpeg which is used in other formula, as well. Perhaps, moving to just pulling from Master, which I realize is not optimal to the project's policy. However, those interested in mpv could get it from another tap, if they're wanting to build it, as well. Thoughts?

**Comment by robertleeblairjr** [comment]
> Got a link?

[https://github.com/mpv-player/mpv/issues/6855](url)

**Comment by robertleeblairjr** [comment]
> EDIT: And what does ffmpeg suggest their packages link to? Just mpv master?

I would have to do a little research to determine if the FFmpeg project has recommendations for mpv specifically. However, I wouldn't think there's a preference since mpv is dependent upon FFmpeg and not vice versa. Here's a list of projects which compile with needing FFmpeg. 
[https://trac.ffmpeg.org/wiki/Projects](url)
The mpv formula doesn't take precedence over the FFmpeg formula though which is used and a requirement in other Homebrew-Core formula.

**Comment by robertleeblairjr** [comment]
> Got a link?
> 
> EDIT: And what does ffmpeg suggest their packages link to? Just mpv master?

Maintainers in this issue suggested using the cask if mpv will not build against FFmpeg 4.2. Unless, homebrew can make an exception to allow pulling master, instead of the much older source tarball, then this will stay broken for some time. As of right now, the two major choices to allow the new version of FFmpeg formula is to make the exception for mpv formula or remove it since there is a cask. Any other ideas?
[https://github.com/mpv-player/mpv/issues/6850](url)

**Comment by pigoz** [comment]
mpv builds with FFmpeg 4.2, but in general if you build mpv with any version of FFmpeg, it is ABI compatible only with that version (which causes crashed if you try to use dylibs from another FFmpeg version).

I suggest you use a [revision](https://rubydoc.brew.sh/Formula#revision%3D-class_method) to force a rebuild of the formula against the new FFmpeg. That's what other maintainers do.

**Comment by robertleeblairjr** [comment]
> mpv builds fine with FFmpeg 4.2, but in general if you build mpv with any version of FFmpeg, it is ABI compatible only with that version (which can cause crashes if you try to use dylibs from another FFmpeg version).
> 
> I suggest you use a [revision](https://rubydoc.brew.sh/Formula#revision%3D-class_method) to force a rebuild of the formula against the new FFmpeg. That's what other package maintainers do.

Sounds good and I've learned something new. However it sounds like a catch-22 to me. How do you have FFmpeg with it's linked libraries committed to the repo in order to revise mpv so that it will build against the newly committed FFmpeg if it can't be committed because of failing a link test with mpv? Or, how do you have mpv rebuild whenever an FFmpeg PR needs checked/tested for commit?

**Comment by pigoz** [comment]
You could add a separate commit in the same PR that bumps the revision. I'm not familiar that much with homebrew, but I would imagine whatever tests run, they do on the changeset and not the single commits.

**Comment by robertleeblairjr** [comment]
> You could add a separate commit in the same PR that bumps the revision. I'm not familiar that much with homebrew, but I would imagine whatever tests run, they do on the changeset and not the single commits.

Neither am I. But, that's a great suggestion if it's feasible. I'll see what I come across as to if this is either implemented in other formula and is doable.

**Comment by robertleeblairjr** [comment]
> Created with `brew bump-formula-pr`.

Can you close/reopen to initiate again or comment to @BrewTestBot ? I've submitted a PR for mpv to bump the revision from 3 > 4. Unsure if that's all which was needed in order to get this PR to pass mpv link test. But, giving it a shot.

**Comment by jonchang** [comment]
A separate pull request to bump the `revision` of mpv is not needed, since it will still be building against ffmpeg 4.1.x. The revision bump should instead happen in this pull request so that ffmpeg 4.2 + mpv revision are built and tested at the same time.

Furthermore ffmpeg 4.2 is failing `brew audit` so none of that will happen until the audit is fixed.

**Comment by chenrui333** [comment]
Sure, restart a build now.

**Comment by chenrui333** [comment]
@BrewTestBot test this please

**Comment by robertleeblairjr** [comment]
> A separate pull request to bump the `revision` of mpv is not needed, since it will still be building against ffmpeg 4.1.x. The revision bump should instead happen in this pull request so that ffmpeg 4.2 + mpv revision are built and tested at the same time.
> 
> Furthermore ffmpeg 4.2 is failing `brew audit` so none of that will happen until the audit is fixed.

Thanks. Yep, I can't do that in this PR. Also, didn't know it was failing because of audit. Thought it was just a link test fail with mpv and caused by python.
FFmpeg Dependencies from `brew deps --installed --tree`

> ffmpeg
|── aom
|── fontconfig
│   └── freetype
│       └── libpng
|── freetype
│   └── libpng
|── frei0r
|── gnutls
│   |── gmp
│   |── libidn2
│   │   |── gettext
│   │   └── libunistring
│   |── libtasn1
│   |── libunistring
│   |── nettle
│   │   └── gmp
│   |── p11-kit
│   │   └── libffi
│   └── unbound
│       |── libevent
│       │   └── openssl
│       └── openssl
|── lame
|── libass
│   |── freetype
│   │   └── libpng
│   |── fribidi
│   └── harfbuzz
│       |── cairo
│       │   |── fontconfig
│       │   │   └── freetype
│       │   │       └── libpng
│       │   |── freetype
│       │   │   └── libpng
│       │   |── glib
│       │   │   |── gettext
│       │   │   |── libffi
│       │   │   |── pcre
│       │   │   └── python
│       │   │       |── gdbm
│       │   │       |── openssl
│       │   │       |── readline
│       │   │       |── sqlite
│       │   │       │   └── readline
│       │   │       └── xz
│       │   |── libpng
│       │   |── lzo
│       │   └── pixman
│       |── freetype
│       │   └── libpng
│       |── glib
│       │   |── gettext
│       │   |── libffi
│       │   |── pcre
│       │   └── python
│       │       |── gdbm
│       │       |── openssl
│       │       |── readline
│       │       |── sqlite
│       │       │   └── readline
│       │       └── xz
│       |── graphite2
│       └── icu4c
|── libbluray
│   |── fontconfig
│   │   └── freetype
│   │       └── libpng
│   └── freetype
│       └── libpng
|── libsoxr
|── libvorbis
│   └── libogg
|── libvpx
|── opencore-amr
|── openjpeg
│   |── libpng
│   |── libtiff
│   │   └── jpeg
│   └── little-cms2
│       |── jpeg
│       └── libtiff
│           └── jpeg
|── opus
|── rtmpdump
│   └── openssl
|── rubberband
│   |── libsamplerate
│   │   └── libsndfile
│   │       |── flac
│   │       │   └── libogg
│   │       |── libogg
│   │       └── libvorbis
│   │           └── libogg
│   └── libsndfile
│       |── flac
│       │   └── libogg
│       |── libogg
│       └── libvorbis
│           └── libogg
|── sdl2
|── snappy
|── speex
│   └── libogg
|── tesseract
│   |── leptonica
│   │   |── giflib
│   │   |── jpeg
│   │   |── libpng
│   │   |── libtiff
│   │   │   └── jpeg
│   │   |── openjpeg
│   │   │   |── libpng
│   │   │   |── libtiff
│   │   │   │   └── jpeg
│   │   │   └── little-cms2
│   │   │       |── jpeg
│   │   │       └── libtiff
│   │   │           └── jpeg
│   │   └── webp
│   │       |── jpeg
│   │       |── libpng
│   │       └── libtiff
│   │           └── jpeg
│   └── libtiff
│       └── jpeg
|── theora
│   |── libogg
│   └── libvorbis
│       └── libogg
|── x264
|── x265
|── xvid
└── xz

**Comment by jonchang** [comment]
Feel free to open a separate pull request if you can fix the audit and reverse dependency issues.

**Comment by robertleeblairjr** [comment]
> Feel free to open a separate pull request if you can fix the audit and reverse dependency issues.

Merely a novice hobbyist trying to help solve problems. So, my input is limited but I may be able to find more simple fixes. Looks like there are two libraries which build using python in ffmpeg; libass and glib. So, I wouldn't know what has changed to cause this issue from the last 4.1.4 > 4.2. Obviously, if these same dependencies which are used by the ffmpeg project had failed, then they would not have released this version. What does audit explicitly do which isn't allowed but is fine with building from source?

**Comment by jonchang** [comment]
Audit performs automated checks to ensure that submitted formula comply with various Homebrew policies. If you applied this pull request to your local install and ran `brew install -s ffmpeg && brew audit ffmpeg` you would likely get the same result as our CI infrastructure.

**Comment by robertleeblairjr** [comment]
The audit installs gem bundler for dependencies. So, if python being installed by a library wasn't a problem before, then why would it be now? Would a dependency of the library being built need to be stated for the formula as just an option using :build?

**Comment by robertleeblairjr** [comment]
> Audit performs automated checks to ensure that submitted formula comply with various Homebrew policies. If you applied this pull request to your local install and ran `brew install -s ffmpeg && brew audit ffmpeg` you would likely get the same result as our CI infrastructure.

Alright, I'll look at if anything has changed/added by ffmpeg project for building. Is it possible that something that was changed from 4.1.4 > 4.2 is fine for building from source but now has caused audit to fail?

**Comment by jonchang** [comment]
> The audit installs gem bundler for dependencies. So, if python being installed by a library wasn't a problem before, then why would it be now? Would a dependency of the library being built need to be stated for the formula as just an option using :build?

I don't understand what this means, or how bundler is related to this audit message at all.

```
Error: 1 problem in 1 formula detected
ffmpeg:
  * Non-executables were installed to "/usr/local/opt/ffmpeg/bin"
    The offending files are:
      /usr/local/opt/ffmpeg/bin/python
```

This formula is installing a non-executable file called `python` into the `bin` folder. Therefore, the solution should involve not placing non-executables into the `bin` folder.

**Comment by robertleeblairjr** [comment]
> > The audit installs gem bundler for dependencies. So, if python being installed by a library wasn't a problem before, then why would it be now? Would a dependency of the library being built need to be stated for the formula as just an option using :build?
> 
> I don't understand what this means, or how bundler is related to this audit message at all.
> 
> ```
> Error: 1 problem in 1 formula detected
> ffmpeg:
>   * Non-executables were installed to "/usr/local/opt/ffmpeg/bin"
>     The offending files are:
>       /usr/local/opt/ffmpeg/bin/python
> ```
> 
> This formula is installing a non-executable file called `python` into the `bin` folder. Therefore, the solution should involve not placing non-executables into the `bin` folder.

Yes, I get that. Should I assume this was not happening in 4.1.4 and this is only caused by ffmpeg? That was why I was asking. I don't know all of the details about audit and what's allowed and isn't. Perhaps, this is acceptable for just building from source but not when brew checks for dependencies and this one isn't declared? Maybe it needs to be declared as an option such as :build or :test are used for other formula?

**Comment by jonchang** [comment]
> Yes, I get that. Should I assume this was not happening in 4.1.4 and this is only caused by ffmpeg? That was why I was asking.

I don't know. You can test this out yourself by running `brew install -s` and `brew audit`.

> I don't know all of the details about audit and what's allowed and isn't.

`brew audit` will tell you what is allowed and what isn't.

> Perhaps, this is acceptable for just building from source but not when brew checks for dependencies and this one isn't declared?

Formula which fail `brew audit` are generally not allowed in homebrew-core, so no, this wouldn't be acceptable as-is. See https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request

> Maybe it needs to be declared as an option such as :build or :test are used for other formula?

I don't know what "it" is referring to in this sentence.

As a general note - I'm going to need you to start actually working on this yourself. I'm willing to help you through this process but I'm going to need to see code or a pull request before responding to further questions about this.

**Comment by robertleeblairjr** [comment]
> > Yes, I get that. Should I assume this was not happening in 4.1.4 and this is only caused by ffmpeg? That was why I was asking.
> 
> I don't know. You can test this out yourself by running `brew install -s` and `brew audit`.
> 
> > I don't know all of the details about audit and what's allowed and isn't.
> 
> `brew audit` will tell you what is allowed and what isn't.
> 
> > Perhaps, this is acceptable for just building from source but not when brew checks for dependencies and this one isn't declared?
> 
> Formula which fail `brew audit` are generally not allowed in homebrew-core, so no, this wouldn't be acceptable as-is. See https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request
> 
> > Maybe it needs to be declared as an option such as :build or :test are used for other formula?
> 
> I don't know what "it" is referring to in this sentence.
> 
> As a general note - I'm going to need you to start actually working on this yourself. I'm willing to help you through this process but I'm going to need to see code or a pull request before responding to further questions about this.



> > Yes, I get that. Should I assume this was not happening in 4.1.4 and this is only caused by ffmpeg? That was why I was asking.
> 
> I don't know. You can test this out yourself by running `brew install -s` and `brew audit`.
> 
> > I don't know all of the details about audit and what's allowed and isn't.
> 
> `brew audit` will tell you what is allowed and what isn't.
> 
> > Perhaps, this is acceptable for just building from source but not when brew checks for dependencies and this one isn't declared?
> 
> Formula which fail `brew audit` are generally not allowed in homebrew-core, so no, this wouldn't be acceptable as-is. See https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request
> 
> > Maybe it needs to be declared as an option such as :build or :test are used for other formula?
> 
> I don't know what "it" is referring to in this sentence.
The binary being installed, which references python, that's used during the check.

> 
> As a general note - I'm going to need you to start actually working on this yourself. I'm willing to help you through this process but I'm going to need to see code or a pull request before responding to further questions about this.

You're talking to someone, like many people who use homebrew, who know little to no "coding" of formula they use to include the project. To top it off, if their primary occupation isn't even related, then this doesn't help the foundation of their knowledge to approach or solve problems the same as a dev or other person who's daily activities are centered around software. So, I'm guessing this will stay a PR for quite some time unless it's an actual simple remedy that someone spots, FFmpeg project has another update which "corrects" this, or some other difference between between the current 4.1.4 > 4.2 that's caused audit to fail. In my free time, I'll give it a go and see what else I could contribute.

**Comment by retokromer** [comment]
@chenrui333 Could you possibly add the bump revision of `mpv` commit here in this PR?

**Comment by chenrui333** [comment]
Sure, doing that now.

**Comment by chenrui333** [comment]
Still failing:

```
16:09:36 /private/tmp/mpv-20190810-80715-1gwap3d/mpv-0.29.1/video/out/cocoa_cb_common.swift:337:39: error: cannot call value of non-function type '[NSScreen]'
16:09:36         let screens = NSScreen.screens()
16:09:36                       ~~~~~~~~~~~~~~~~^~
16:09:36                                       
16:09:36 
16:09:36 Waf: Leaving directory `/private/tmp/mpv-20190810-80715-1gwap3d/mpv-0.29.1/build'
16:09:36 Build failed
16:09:36  -> task in 'osdep/macOS_swift.o osdep/macOS_swift.h osdep/macOS_swift.swiftmodule' failed with exit status 1 (run with -v to display more information)
16:09:36 
```

**Comment by retokromer** [comment]
The “python” issue can be resolved either upstream by avoiding the additional `python` folder created in http://git.ffmpeg.org/gitweb/ffmpeg.git/commit/50e194e6e126b7fe8e91ecc2e929335d4dc64fcf or here by modifying:
https://github.com/Homebrew/homebrew-core/blob/0784f5515cd426750d14bfae5e77f3ca00f44c7f/Formula/ffmpeg.rb#L92 
in order to install `convert.py` and `convert_from_tensorflow.py` directly on the `bin` level rather than into `bin/python`.

**Comment by JCount** [comment]
The `v0.29.1` release of `mpv` will not build using any Xcode release after `10.1`. This is due to the removal of support for `Swift 3` in the `10.2` release. 

See: [**Swift 5 Release Notes for Xcode 10.2**](https://developer.apple.com/documentation/xcode_release_notes/xcode_10_2_release_notes/swift_5_release_notes_for_xcode_10_2)
```
Swift 3 mode has been removed. Supported values for the -swift-version flag are
4, 4.2, and 5. (43101816)
```

[The code has been migrated to support Swift 5 with Swift 4 as a fallback, as well as dynamic linking of the libs.](https://github.com/mpv-player/mpv/commit/a8c2e2986838dccbcc)

However, these changes currently only exist on the master branch of `mpv` and are not included in any tags. The head spec builds and works fine for me on my Mac running 10.14.16, after being built with Xcode 10.3 build 10G8.

This puts the maintainers in a bit of a sticky situation, when a dependency breaks without an easy fix but that is par for the course.

One less than ideal solution that has limited precedent would be simply distribute the bottle that is generated for High Sierra to Mojave as well. This would probably require some manual testing, but unless things have changed significantly since I was a maintainer, it should work.

Whether you want to go there is another matter entirely, and one I no longer have to spend time considering. 😄

**Comment by retokromer** [comment]
>However, these changes currently only exist on the master branch of `mpv` and are not included in any tags. 

The `mpv` maintainers were asked to make a new release/tag.

**Comment by JCount** [comment]
@retokromer Yes, but they do not plan on doing that as far as I can tell, which means that a creative solution is needed in order for this PR to go through, given that this is an Xcode compatibility issue. In terms of a new tag, they did not seem to be very responsive to the idea, and generally haven't been in the past. You saw that philosophy yourself in the [issue you opened.](https://github.com/mpv-player/mpv/issues/6850)

**Comment by retokromer** [comment]
@JCount I’m lost in the dark. Do you have any glint on how “a creative solution” could look like?

**Comment by chenrui333** [comment]
Refresh the version to 4.2.1

File: Formula/mpv.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class Mpv < Formula
   homepage "https://mpv.io"
   url "https://github.com/mpv-player/mpv/archive/v0.29.1.tar.gz"
   sha256 "f9f9d461d1990f9728660b4ccb0e8cb5dce29ccaa6af567bec481b79291ca623"
-  revision 3
+  revision 4
```
Comment (retokromer) [inline Formula/mpv.rb]: See https://github.com/Homebrew/homebrew-core/pull/42827#issuecomment-521474699

File: Formula/ffmpeg.rb [hunk 2]
```diff
@@ -1,9 +1,8 @@
 class Ffmpeg < Formula
   desc "Play, record, convert, and stream audio and video"
   homepage "https://ffmpeg.org/"
-  url "https://ffmpeg.org/releases/ffmpeg-4.1.4.tar.xz"
-  sha256 "f1f049a82fcfbf156564e73a3935d7e750891fab2abf302e735104fd4050a7e1"
-  revision 1
+  url "https://ffmpeg.org/releases/ffmpeg-4.2.1.tar.xz"
+  sha256 "cec7c87e9b60d174509e263ac4011b522385fd0775292e1670ecc1180c9bb6d4"
```
Comment (retokromer) [inline Formula/ffmpeg.rb]: See https://github.com/Homebrew/homebrew-core/pull/42827#issuecomment-520226782

**Comment by danielbair** [comment]
> ```
> Error: 1 problem in 1 formula detected
> ffmpeg:
>   * Non-executables were installed to "/usr/local/opt/ffmpeg/bin"
>     The offending files are:
>       /usr/local/opt/ffmpeg/bin/python
> ```

I am wondering if the python directory and the two python scripts inside that directory should just be moved from bin/ to share/ or something, because for those python scripts to work it requires a "pip install tensorflow" and that requires homebrew formula "hdf5" for the python "tensorflow" module and its python dependencies to install correctly.

I have solved the "brew audit --strict" over on danielbair/tap/ffmpeg but I am considering the solution I have suggested here might be better.

**Comment by retokromer** [comment]
It’s exactly what I meant to say in https://github.com/Homebrew/homebrew-core/pull/42827#issuecomment-520226782. Sorry for being unclear! I am not English native…

**Comment by danielbair** [comment]
> It’s exactly what I meant to say in [#42827 (comment)](https://github.com/Homebrew/homebrew-core/pull/42827#issuecomment-520226782). Sorry for being unclear! I am not English native…

Ok, I can't publish to this pull request, but I do have a fix.

**Comment by danielbair** [comment]
This is my proposed solution:

The python directory and the two python scripts inside that directory are moved from bin/ to pkgshare/ because the python scripts require python module "tensorflow" and that requires homebrew formula "hdf5" to install correctly. Being as the ffmpeg make file doesn't do this itself, it seems that these files are at most experimental or at least examples.

```
@ -92,6 +92,9 @@ class Ffmpeg < Formula
    # Build and install additional FFmpeg tools
    system "make", "alltools"
    bin.install Dir["tools/*"].select { |f| File.executable? f }
+
+   # Fix for Non-executables that were installed to bin/
+   mv bin/"python", pkgshare/"python", :force => true
  end

  test do
```

**Comment by retokromer** [comment]
>I can't publish to this pull request, but I do have a fix.

@danielbair That’s my problem too, since almost one month… Thanks for the proposed patch!

**Comment by retokromer** [comment]
The patch proposed by @danielbair works well! After adding
```rb
mv bin/"python", pkgshare/"python", :force => true
```
the formula works without any error or warning on my end. Could possibly @chenrui333 (or any other person who can add a commit here) apply it? Thanks a lot!

**Comment by chenrui333** [comment]
@retokromer, getting back on this and commit the change now.

**Comment by chenrui333** [comment]
@retokromer @danielbair is [this commit](https://github.com/Homebrew/homebrew-core/pull/42827/commits/492ef1bcfabbb296dc3c643a14ccf94fd497bc3f) needed for the formula?

**Comment by retokromer** [comment]
@chenrui333 As there are no longer options here, I guess it’s better to add this dependence.

**Comment by chenrui333** [comment]
Sounds good!

**Comment by chenrui333** [comment]
looks like the build still failing.

**Comment by retokromer** [comment]
I guess this is still the `mpv` issue mentioned in https://github.com/Homebrew/homebrew-core/pull/42827#issuecomment-521474699.

**Comment by chenrui333** [comment]
yeah, exact that swift issue.

**Review by danielbair — CHANGES_REQUESTED** [review]
If the revision change to the mpv.rb formula were removed from this pull request, the ffmpeg.rb formula might just pass the checks now.

**Review by danielbair — APPROVED** [review]
This is an advantage to include and it is not causing the build to fail.

**Review by danielbair — APPROVED** [review]
This passes the brew audit tests on my system

**Review by danielbair — CHANGES_REQUESTED** [review]
If the revision change to the mpv.rb formula were removed from this pull request, the ffmpeg.rb formula might just pass the checks now.

**Comment by danielbair** [comment]
Ok,

This is my proposed fix for the mpv.rb formula. It does build fine on the latest xcode and with the latest ffmpeg 4.2.1 and passes "brew audit --strict' on my system.

```
class Mpv < Formula
  desc "Media player based on MPlayer and mplayer2"
  homepage "https://mpv.io"
- url "https://github.com/mpv-player/mpv/archive/v0.29.1.tar.gz"
- sha256 "f9f9d461d1990f9728660b4ccb0e8cb5dce29ccaa6af567bec481b79291ca623"
- revision 4
+ url "https://github.com/mpv-player/mpv/archive/6e6ec331685c78584a818f524286670911e8b4af.tar.gz"
+ version "0.29.2"
+ sha256 "fbb5ebc72c55af6e62cb3835b87b0fd26160533350f17e73712791870bdbe017"
  head "https://github.com/mpv-player/mpv.git"

  bottle do
```

**Comment by slhck** [comment]
I don't believe that it is acceptable to “invent” a release just to make a Homebrew formula build. There is simply no 0.29.2 version of `mpv` yet, so we can't put that in a Homebrew formula and cut from master whenever we feel like.

Note that I am absolutely not against finding a solution here, but I strongly believe that this is not a good one. Having an exception for the formula to pull from master while keeping its old version (and providing a caveat) seems more meaningful to me.

**Comment by danielbair** [comment]
> I don't believe that it is acceptable to “invent” a release just to make a Homebrew formula build. There is simply no 0.29.2 version of `mpv` yet, so we can't put that in a Homebrew formula and cut from master whenever we feel like.
> 
> Note that I am absolutely not against finding a solution here, but I strongly believe that this is not a good one. Having an exception for the formula to pull from master while keeping its old version (and providing a caveat) seems more meaningful to me.

I tried keeping the version the same as before, but brew audit --strict failed complaining that the source and sha had changed for the same version...

I do understand about not inventing a version, because if the next tag comes out with that same version number then there is a conflict.

Anyone else know what to do here?

**Comment by SMillerDev** [comment]
@Homebrew/core opinions?

**Comment by SMillerDev** [comment]
Could you rebase on master?

**Comment by danielbair** [comment]
> Could you rebase on master?

Do you mean https://github.com/mpv-player/mpv/archive/master.tar.gz ?
But the sha will always be changing won't it?

**Comment by slhck** [comment]
No, Sean meant rebasing your PR against `homebrew-core`'s master branch.

**Comment by SMillerDev** [comment]
> No, Sean meant rebasing the PR against homebrew-core's master branch.

Yeah, this

**Comment by danielbair** [comment]
@chenrui333 you will have to rebase this PR against the current master, because of its age.

**Comment by jonchang** [comment]
Fixed by https://github.com/Homebrew/homebrew-core/pull/44282

**Pull request closed without merging**